src/gui/sdl_mapper.cpp

```diff
@@ -2990,37 +2993,67 @@ void MAPPER_DisplayUI() {
 	GFX_EndUpdate( nullptr );
 	mapper.window = GFX_GetWindow();
 	if (mapper.window == nullptr) {
-		E_Exit("Could not initialize video mode for mapper: %s", SDL_GetError());
+		E_Exit("MAPPER: Could not initialize video mode: %s",
+		       SDL_GetError());
 	}
 	mapper.renderer = SDL_GetRenderer(mapper.window);
 #if C_OPENGL
 	SDL_GLContext context = nullptr;
 	if (!mapper.renderer) {
 		context         = SDL_GL_GetCurrentContext();
-		mapper.renderer = SDL_CreateRenderer(mapper.window, -1, 0);
+		if (!context) {
+			E_Exit("MAPPER: Failed to retrieve current OpenGL context: %s",
+			       SDL_GetError());
+		}
+
+		const auto renderer_drivers_count = SDL_GetNumRenderDrivers();
+		if (renderer_drivers_count < 0) {
+			E_Exit("MAPPER: Failed to retrieve available SDL renderer drivers: %s",
+			       SDL_GetError());
+		}
+		int renderer_driver_index = -1;
+		for (int i = 0; i < renderer_drivers_count; i++) {
```

Comment:
What's the difference specifically in this case anyway?

Reply:
It's technically a micro-optimization and suggested by PVS Studio in many places (they do a full blown analysis and performance comparison here: https://pvs-studio.com/en/blog/posts/cpp/0093/)

But it's also a Staging coding style whenever we want to increment a value or iterator in a single statement, without any side effects.

Basically when an increment happens on it's own:

```c++
++i;
i++;
i = i + 1;
```

All of those can be boiled down to `++i`. The `; i++)` in the for loop is similar.. it's stand-alone statement that's run at the end of the for loop pass, so it doesn't matter if it's `; i++)`, `; i = i + 1)`, or `++i)`.

The place to use the post-increment is when you want to read the original value /before/ doing the increment.

```c++
int i = 0;
while (i++ != 1) {
  // this will loop once, i is still 0
  assert (i == 0);
}
```

Because `i++` is expanded to "let i be read first, followed by incrementing the i after the statement. So the compile sees it as:

```c++
int i = 0;
while (i != 1) {
  // this will loop once
  i = i + 1;
}
```

Where as `++i` does it right then and there:

```c++
int i = 0;
while (++i != 1) {
  // this won't loop once, i is already 1
}
```

Reply:
Yeah, post-increment/decrement is a source for bugs. Best to train everybody out of using it as a default thing 😅 Of course, use it by all means if the post-operation is warranted.

Reply:
The nice thing with the pre-increment is it simplifies the reading. You can read it as "this variable is one greater" 100% of the time.

Where as post-increment, you've gotta think through the nuances/side-effects of the value being incremented /after/ the statement is done.